ChaosMod/Effects/db/Player/PlayerTpToEverything.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+	Effect by Gorakh
+*/
+
+#include <stdafx.h>
+
+static void OnTick()
+{
+	std::vector<Entity> allEntities = {};
+	for (Ped ped : GetAllPeds())
+	{
+		if (!IS_PED_A_PLAYER(ped))
+		{
+			allEntities.push_back(ped);
+		}
+	}
+
+	Vehicle playerVeh = IS_PED_IN_ANY_VEHICLE(PLAYER_PED_ID(), false) ? GET_VEHICLE_PED_IS_IN(PLAYER_PED_ID(), false) : 0;
+	for (Vehicle veh : GetAllVehs())
+	{
+		if (!playerVeh || veh != playerVeh)
+		{
+			allEntities.push_back(veh);
+		}
+	}
+
+	for (Object prop : GetAllProps())
+	{
+		allEntities.push_back(prop);
+	}
+
+	if (!allEntities.empty())
+	{
+		Entity randomEntity = allEntities[g_Random.GetRandomInt(0, allEntities.size() - 1)];
+		TeleportPlayer(GET_ENTITY_COORDS(randomEntity, true));
+	}
+
+	WAIT(g_Random.GetRandomInt(5000, 8000));
```

Comment:
Please change to a local variable and check on start of the onTick (see the airstrike effect)

Reply:
Fixed :)